src/ios/PurchasesPlugin.swift

```diff
@@ -1,10 +1,397 @@
+        simpleStringSet(command: command, name: "onesignalID", setFunction: RCCommonFunctionality.setOnesignalID)
+    }
+
+    @objc(setAirshipChannelID:) func setAirshipChannelID(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "airshipChannelID", setFunction: RCCommonFunctionality.setAirshipChannelID)
+    }
+
+    @objc(setMediaSource:) func setMediaSource(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "mediaSource", setFunction: RCCommonFunctionality.setMediaSource)
+    }
+
+    @objc(setCampaign:) func setCampaign(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "campaign", setFunction: RCCommonFunctionality.setCampaign)
+    }
+
+    @objc(setAdGroup:) func setAdGroup(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "adGroup", setFunction: RCCommonFunctionality.setAdGroup)
+    }
+
+    @objc(setAd:) func setAd(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "ad", setFunction: RCCommonFunctionality.setAd)
+    }
+
+    @objc(setKeyword:) func setKeyword(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "keyword", setFunction: RCCommonFunctionality.setKeyword)
+    }
+
+    @objc(setCreative:) func setCreative(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "creative", setFunction: RCCommonFunctionality.setCreative)
+    }
+
+    @objc(collectDeviceIdentifiers:) func collectDeviceIdentifiers(command: CDVInvokedUrlCommand) {
+        RCCommonFunctionality.collectDeviceIdentifiers()
+        self.sendOKFor(command: command)
+    }
+
+    @objc(canMakePayments:) func canMakePayments(command: CDVInvokedUrlCommand) {
+        guard let features = command.arguments[0] as? NSArray as? [NSNumber] else {
+            self.sendBadParameterFor(command: command, parameterNamed: "features", expectedType: NSArray.self)
+            return
+        }
+
+        let canMakePayments = RCCommonFunctionality.canMakePayments(withFeatures: features)
+        let result = CDVPluginResult(status: .ok, messageAs: canMakePayments)
+        self.commandDelegate.send(result, callbackId: command.callbackId)
+    }
+
+}
+
+// Currently unused
+extension CDVPurchasesPlugin {
+
+    func makeDeferredPurchase(command: CDVInvokedUrlCommand) {
+        let callbackID = command.arguments[0] as! NSNumber
+        assert(callbackID.intValue >= 0)
+        let defermentBlock = self.defermentBlocks[callbackID.intValue]
+        RCCommonFunctionality.makeDeferredPurchase(defermentBlock,
+                                                   completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+}
+
+extension CDVPurchasesPlugin: PurchasesDelegate {
+
+    private func purchases(_ purchases: Purchases, didReceiveUpdated purchaserInfo: Purchases.PurchaserInfo) {
```

Comment:
I didn't know you could do this - i.e.: conform to the protocol but have the methods be private. Is there any advantage to it, though? like, since the conformance is public, should we just have the methods have the same privacy level?

Reply:
I hate how the methods are public, nobody should call them except the framework.